api/ws/connector.js

```diff
@@ -95,17 +115,30 @@ function Websockets(conf, logger) {
             connection.sendUTF(JSON.stringify(initMessageObject));
             connection.on('close', function() {
                 me.logger.info("Websocket connection closed.");
-                me.reconnect();
+                setTimeout(function() {
+                    me.reconnect();
+                }, parseInt(me.retryTime));
             });
             connection.on('message', function(message) {
-                me.logger.info('Fired STATUS: ', message.utf8Data);
-                me.onMessage(message.utf8Data);
+                parseMessage(message.utf8Data, function(err, messageObject) {
+                    if(!err) {
+                        if(messageObject.code === 200) {
+                            me.retryTime = conf.connector.ws.retryTime;
+                            me.logger.info('WSConnector: Connection successful to: ' + conf.connector.ws.host + ':' + conf.connector.ws.port);
+                        } else if(messageObject.code === 1024) {
+                            me.logger.info('Fired STATUS: ', messageObject.content);
+                            me.onMessage(messageObject.content);
+                        } else if(messageObject.code === 500 || messageObject.code === 401) {
+                            me.calculateRetryTime(messageObject.content);
+                        }
+                    }
```

Comment:
there is missing else clause(for case when err !== null). Add it with logger.error message